sdk/core/azure-core/src/main/java/com/azure/core/http/policy/BearerTokenAuthenticationPolicy.java

```diff
@@ -198,4 +221,24 @@ private void setAuthorizationHeaderHelperSync(HttpPipelineCallContext context,
     private static void setAuthorizationHeader(HttpHeaders headers, String token) {
         headers.set(HttpHeaderName.AUTHORIZATION, BEARER + " " + token);
     }
+
+    private TokenRequestContext getTokenRequestContextForCaeChallenge(HttpResponse response) {
+        String decodedClaims = null;
+        String encodedClaims = AuthorizationChallengeParser.getChallengeParameterFromResponse(response, "Bearer", "claims");
+
+        try {
+            if (!CoreUtils.isNullOrEmpty(encodedClaims)) {
+                decodedClaims = new String(Base64.getDecoder().decode(encodedClaims));
+            }
+        } catch (IllegalArgumentException e) {
+            // We don't want to throw here, but we want to log this for future incident investigation.
+            LOGGER.warning("Failed to decode the claims from the CAE challenge. Encoded claims" + encodedClaims);
```

Comment:
```suggestion
            LOGGER.warning("Failed to decode the claims from the CAE challenge. Encoded claims: " + encodedClaims);
```

To make it easier to read